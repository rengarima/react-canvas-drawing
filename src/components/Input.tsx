import React, {useState} from "react";
import {InputProps} from "../types/InputProps";

export const Input = ({
    id, placeholder, icon, error, type = "text",executeInput

                      }:InputProps): JSX.Element => {
    const dangerStyle = error ? " is-danger": "";

    const [value, setValue ] = useState();

    function handleChange(e: any): void {
        const inputVal = e.target.value;
        setValue(inputVal);

    }

    const handleOnKeyPress = (e:any) =>{
        e.key === "Enter" && executeInput(value);
    }

    return (
        <div className="columns">
            <div className="column is-fullhd" >
                <p className="control has-icons-left">
                    <input aria-label={id}
                        className={"input is-rounded" + dangerStyle } type={type}
                           placeholder={placeholder}
                           onKeyPress={handleOnKeyPress}
                           onChange={(value: any) => {
                                handleChange(value);
                            }}/>
                    <span className="icon is-small is-left">
                      <i className={icon}></i>
                    </span>
                </p>
                { error && <p
                    className="help is-danger"
                    style={{ fontSize: 15, overflowWrap: "normal" }}
                >
                    &nbsp;{error + ""}
                </p> }
            </div>
        </div>
    );
}
