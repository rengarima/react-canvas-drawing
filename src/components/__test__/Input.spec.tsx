import React from 'react';
import { render, fireEvent, RenderResult } from '@testing-library/react';
import {Input} from '../Input';

let documentBody: RenderResult;
describe('<Input />', () => {
    beforeEach(() => {
        documentBody = render(<Input
            id="test-input"
            placeholder="Enter Command"
            icon="icon"
        />);
    });
    it('Input is rendered', () => {
        expect(documentBody.getByLabelText("test-input")).toBeDefined();
        expect(documentBody.getByPlaceholderText("Enter Command")).toBeDefined();
    });

    it('Input is rendered with Error', () => {
        expect(documentBody.getByLabelText("test-input")).toBeDefined();
        expect(documentBody.getByPlaceholderText("Enter Command")).toBeDefined();
    });

    it('Value is changed', () => {
        const input = documentBody.getByLabelText("test-input");
        fireEvent.change(input, {target: {value: "C 10 4"}});
        //@ts-ignore
        expect(input.value).toBe("C 10 4");
    });

    it('Input is rendered with Error', () => {
        const documentBodyWithError = render(<Input
            id="test-input"
            placeholder="Enter Command"
            icon="icon"
            error="ERROR"
        />);
        expect(documentBodyWithError.getByText("ERROR")).toBeDefined();
    });

    it("matches snapshot", () => {
        const { baseElement } = documentBody;
        expect(baseElement).toMatchSnapshot();
    });
});
