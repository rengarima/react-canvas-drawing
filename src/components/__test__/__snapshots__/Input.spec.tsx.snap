// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="columns"
    >
      <div
        class="column is-fullhd"
      >
        <p
          class="control has-icons-left"
        >
          <input
            aria-label="test-input"
            class="input is-rounded"
            placeholder="Enter Command"
            type="text"
          />
          <span
            class="icon is-small is-left"
          >
            <i
              class="icon"
            />
          </span>
        </p>
      </div>
    </div>
  </div>
</body>
`;
